src/components/HomeSearchBox/index.tsx

```diff
@@ -20,23 +40,63 @@ const HomeSearchBox = () => {
 
   const handleKeyPress = (event: React.KeyboardEvent<HTMLInputElement>) => {
     if (event.key === 'Enter') {
+      setActive(0)
+      setIsShow(false)
+      setSearchValue(filtered[active])
       handleSearchClick()
+    } else if (event.key === 'Up') {
+      return active === 0 ? null : setActive(active - 1)
+    } else if (event.key === 'Down') {
+      return active - 1 === filtered.length ? null : setActive(active + 1)
+    }
+  }
+
+  const renderAutocomplete = () => {
+    if (isShow && searchValue) {
+      if (filtered.length) {
+        return (
+          <ul className="home-search__box__autocomplete">
+            {filtered.map((suggestion, index) => {
+              let className
+              if (index === active) {
+                className = 'active'
+              }
+              return (
+                <li className={className} key={suggestion} onClick={handleListClick}>
+                  {suggestion}
+                </li>
+              )
+            })}
+          </ul>
```

Comment:
This generating element function could be moved into a helper function for this specific function? 🤓